### PR TITLE
Added: cart management

### PR DESCRIPTION
**Pull request by poornima-alugubelly**
Added:
Add to cart
Cart increment decrement product
Refactored file structure

### CODE REVIEW

File: src/context/cartContext.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	});
+
+	const { SET_CART } = actionTypes;
+	const { auth } = useAuth();
+	useEffect(
+		() =>
+			(async () => {
+				console.log("in use effect");
+				if (auth.isAuth) {
+					console.log("calling");
```
Comment (AbhishekkGautam) [inline src/context/cartContext.js:21]: We should remove console.log

File: src/context/cartContext.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+					const cart = await getCartService(auth.token);
+					console.log(cart);
+					if (cart) cartDispatch({ type: SET_CART, payload: { cart } });
+				}
+			})(),
+		[auth.isAuth]
+	);
+
+	console.log("cart", ...cartState.cart);
+	return (
```
Comment (AbhishekkGautam) [inline src/context/cartContext.js:31]: Remove console.log

File: src/pages/ProductListing/components/ProductCard/ProductCard.jsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+	};
+	const navigate = useNavigate();
+	const goToCart = () => navigate("/cart");
+	useEffect(() => {
+		cartState.cart.find((item) => item._id === product._id) && setInCart(true);
+	}, [cartState.cart]);
+	return (
+		<div class="card card-vertical">
+			<div class="img-container">
+				<img src={product.src} alt="product image" class="img-responsive" />
```
Comment (AbhishekkGautam) [inline src/pages/ProductListing/components/ProductCard/ProductCard.jsx:24]: We can pass `product.name` in alt to have different alt name for product image.

**Review by AbhishekkGautam — APPROVED** [review]
Clean & structured code. Good work 👍

File: src/helpers/filter-functions/category.js [hunk 4]
```diff
@@ -1,4 +1,4 @@
-const getFliteredProducts = (sortedproducts, categories) => {
+export const getFliteredProducts = (sortedproducts, categories) => {
```
Comment (manojasarsa) [inline src/helpers/filter-functions/category.js:1]: You can name `sortedproducts` as `sortedProducts` to be consistent with naming.

File: src/backend/db/users.js [hunk 5]
```diff
@@ -8,31 +8,32 @@ import { formatDate } from "../utils/authUtils";
...
+		createdAt: formatDate(),
+		updatedAt: formatDate(),
+	},
+	{
+		_id: uuid(),
+		firstName: "Adarsh",
+		lastName: "Balak",
+		email: "adarshbalak@gmail.com",
+		password: bcyrpt.hashSync("adarshBalaki123", 5),
+		// eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJfaWQiOiJiYTNjMTRhYS0xYjVlLTQ0Y2UtYTM5Yy01ZWQ0Y2U2NjBjNWUiLCJlbWFpbCI6ImFkYXJzaGJhbGFrQGdtYWlsLmNvbSIsImlhdCI6MTY0NzY3NDk2N30.Ecjwy-pc0mCKxRzgOLGizTRsQzPX05wHvlT6WIlD-Bc
```
Comment (manojasarsa) [inline src/backend/db/users.js:35]: We can remove this comment if not needed.

**Review by manojasarsa — APPROVED** [review]
The code is clean and readable. Good to merge!

**Pull request merged**